src/figures.jl

```diff
@@ -99,6 +99,10 @@ function Base.getindex(fig::Figure, rows, cols, side = GridLayoutBase.Inner())
     fig.layout[rows, cols, side]
 end
 
+function Base.getindex(fig::Figure, index::CartesianIndex{2}, side = GridLayoutBase.Inner())
+    fig.layout[Tuple(index)[1], Tuple(index)[2], side]
+end
```

Comment:
We might also want to implement this logic in [GridLayoutBase](https://github.com/jkrumbiegel/GridLayoutBase.jl), so that indexing for the following things retains the same behaviour:
- `Figure`
- `GridLayout`
- `GridPosition`
- (maybe) `GridSubposition`